tests/unit/amo/components/TestCategory.js

```diff
@@ -161,7 +163,8 @@ describe(__filename, () => {
     const fakeDispatch = sinon.stub(store, 'dispatch');
     render({}, { autoDispatchCategories: false });
 
-    sinon.assert.notCalled(fakeDispatch);
+    sinon.assert.callCount(fakeDispatch, 1);
+    sinon.assert.calledWith(fakeDispatch, setViewContext(fakeCategory.type));
```

Comment:
I saw your question about this test. The test intends to cover [only this if branch](https://github.com/bobsilverberg/addons-frontend/blob/d41b5e99ba24219eab50f1a1127f937497195fa6/src/amo/components/Category/index.js#L95) which is there to prevent fetching data when the data is already loaded. Because of that, I don't think it's necessary to cover the view context logic.

The old approach was to assert that nothing is dispatched but that won't work anymore. I suggest changing the assertion to be specific about category fetching:

```js
import { CATEGORIES_FETCH } from 'core/constants';

sinon.assert.neverCalledWith(fakeDispatch, sinon.match({ type: CATEGORIES_FETCH }));
```

Reply:
Thanks Kumar, that all makes sense. 👍